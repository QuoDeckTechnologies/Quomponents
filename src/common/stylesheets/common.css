/* ================================================= */
/* Container Style */
/* ================================================= */
.qui {
    position: relative;
    overflow: hidden;
    font-family: Arial, Helvetica, sans-serif;
}

/* ================================================= */
/* Float State Definitions*/
/* ================================================= */

.qui.float-none {
    float: none;
    display: block;
}
.qui.float-left {
    float: left;
    display: block;
}
.qui.float-right {
    float: right;
    display: block;
}
.qui.float-inline {
    float: none;
    display: inline-block;
}

/* ================================================= */
/* Layout State Definitions */
/* ================================================= */
.qui.is-hidden {
    display: none;
}
.qui.is-disabled {
    pointer-events: none;
}
.qui.is-loading {
    pointer-events: none;
}
.qui.is-fluid {
    width: 100%;
    min-width: 100%;
}
.qui .is-fluid {
    width: 100%;
    min-width: 100%;
}

/* ================================================= */
/* Font Size Definitions */
/* ================================================= */
.qui .size-massive {
    font-size: 2em;
    line-height: 1.2;
}
.qui .size-huge {
    font-size: 1.5em;
    line-height: 1.2;
}
.qui .size-big {
    font-size: 1.25em;
    line-height: 1.2;
}
.qui .size-normal {
    font-size: 1em;
    line-height: 1.2;
}
.qui .size-small {
    font-size: 0.85em;
    line-height: 1.2;
}
.qui .size-tiny {
    font-size: 0.7em;
    line-height: 1.2;
}

/* ================================================= */
/* Padding Definitions */
/* ================================================= */
.qui .pad-fitted {
    padding: 2px 4px;
}
.qui .pad-compact {
    padding: 4px 8px;
}
.qui .pad-normal {
    padding: 6px 12px;
}
.qui .pad-relaxed,
.qui .pad-normal.size-huge,
.qui .pad-normal.size-massive {
    padding: 10px 20px;
}
.qui .pad-relaxed.size-huge,
.qui .pad-relaxed.size-massive {
    padding: 15px 30px;
}

/* ================================================= */
/* Alignment Definitions */
/* ================================================= */
.qui .left-aligned {
    text-align: left;
}
.qui .right-aligned {
    text-align: right;
}
.qui .center-aligned {
    text-align: center;
}

/* ================================================= */
/* Font Style Definitions */
/* ================================================= */
.qui.qt-lbl,
.qui .qt-lbl {
    font-family: "Oswald Regular";
    font-style: normal;
    font-weight: 400;
    font-size: 11px;
    line-height: 13px;
}
.qui.qt-sm,
.qui .qt-lbl {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 400;
    font-size: 11px;
    line-height: 13px;
}
.qui.qt-mn,
.qui .qt-lbl {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 400;
    font-size: 9px;
    line-height: 9px;
}
.qui.qt-utn,
.qui .qt-lbl {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 400;
    font-size: 10px;
    line-height: 10px;
}

/* ================================================= */
/* Box Shadow Definitions */
/* ================================================= */
.qui.qt-shadow,
.qui .qt-shadow {
    box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
}
