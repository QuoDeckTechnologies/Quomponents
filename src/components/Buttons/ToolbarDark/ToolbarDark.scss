.qui .qui-backbar {
  background-color: #222a35;
  padding: 0.5em 1em;
  text-align: center;
  border-radius: 0.7em;
}
.qui-icon {
  display: flex;
  justify-content: space-around;
}
.qui button {
  border: none;
}
@media screen and (min-width: 320px) {
  .qui .pad-normal {
    padding: 0.32em;
  }
}
