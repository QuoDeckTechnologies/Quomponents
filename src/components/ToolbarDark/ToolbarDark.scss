.qui .qui-toolbar-dark-backbar {
  background-color: #222a35;
  padding: 0.5em 1em;
  text-align: center;
  border-radius: 0.7em;
  .qui.qui-icon-link {
    font-size: 0.8em;
  }
}
.qui-toolbar-dark-icon {
  display: flex;
  justify-content: space-around;
}
@media only screen and (min-width: 320px) and (max-width: 400px) {
  .qui .qui-toolbar-dark-backbar {
    .qui.qui-icon-link {
      font-size: 0.6em;
    }
  }
}
