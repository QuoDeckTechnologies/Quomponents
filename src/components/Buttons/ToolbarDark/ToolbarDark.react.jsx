import React, { useState } from "react";
import PropTypes from "prop-types";
import { motion } from "framer-motion";
import _ from "lodash";
import {
    getQuommons,
    getTranslation,
    getAnimation,
} from "../../../common/javascripts/helpers";

import "@fortawesome/fontawesome-free/css/all.min.css";
import "../../../common/stylesheets/common.css";
import "./ToolbarDark.scss";
import "../../../common/stylesheets/overrule.scss";
import IconLink from "../IconLink/IconLink.react"

ToolbarDark.propTypes = {
    //=======================================
    // Component Specific props
    //=======================================

    /**
   Icon Text has to be in content or passed as children to the component.
   */
    content: PropTypes.arrayOf(PropTypes.shape({
        icon: PropTypes.string,
        label: PropTypes.string,
        link: PropTypes.string,
    })).isRequired,
    /**
    Set action emphasis in increasing order 
    */
    asEmphasis: PropTypes.oneOf(["text", "outlined", "contained"]),
    /**
    Use for rounded corners or circular icon ToolbarDark 
    */
    isCircular: PropTypes.bool,

    // Quommon props
    //=======================================

    /**
    Use to define standard component type
    */
    asVariant: PropTypes.oneOf([
        "primary",
        "secondary",
        "success",
        "warning",
    ]),
    /**
    Use to define component text size in increasing order
    */
    asSize: PropTypes.oneOf([
        "tiny",
        "small",
        "normal",
        "big",
        "huge",
        "massive",
    ]),
    /**
    Use to define component padding in increasing order
    */
    asPadded: PropTypes.oneOf(["fitted", "compact", "normal", "relaxed"]),
    /**
    Use to align content within the component container
    */
    asAligned: PropTypes.oneOf(["left", "right", "center"]),

    /**
    Use to override component colors and behavior
    */
    withColor: PropTypes.shape({
        backgroundColor: PropTypes.string,
        accentColor: PropTypes.string,
        textColor: PropTypes.string,
        hoverBackgroundColor: PropTypes.string,
        hoverTextColor: PropTypes.string,
    }),
    /**
    Use to add an icon to the component
    */
    withIcon: PropTypes.shape({
        icon: PropTypes.string,
        size: PropTypes.string,
        position: PropTypes.oneOf(["center", "left", "right"]),
    }),
    /**
    Use to add a heading label, a footer caption or a title popover to the component
    */
    withLabel: PropTypes.shape({
        format: PropTypes.oneOf(["label", "caption", "popover"]),
        content: PropTypes.string,
        textColor: PropTypes.string,
        hoverTextColor: PropTypes.string,
    }),
    /**
    Use to define the entry animation of the component
    */
    withAnimation: PropTypes.shape({
        animation: PropTypes.oneOf([
            "zoom",
            "collapse",
            "fade",
            "slideDown",
            "slideUp",
            "slideLeft",
            "slideRight",
            ""
        ]),
        duration: PropTypes.number,
        delay: PropTypes.number,
    }),

    /**
    Use to show a translated version of the component text. Dictionary must be valid JSON. 
    */
    withTranslation: PropTypes.shape({
        lang: PropTypes.string,
        tgt: PropTypes.string,
        dictionary: PropTypes.string,
    }),

    /**
    Use to show/hide the component
    */
    isHidden: PropTypes.bool,
    /**
    Use to enable/disable the component
    */
    isDisabled: PropTypes.bool,
    /**
    Use to toggle the component taking the full width of the parent container
    */
    isFluid: PropTypes.bool,

    /**
    ToolbarDark component must have the onClick function passed as props
    */
    onClick: PropTypes.func.isRequired,
};

ToolbarDark.defaultProps = {

    // Component Specific props
    //=======================================
    content: [],
    asEmphasis: "text",
    isCircular: false,

    // Quommon props
    //=======================================
    asVariant: "primary",
    asSize: "normal",
    asPadded: "normal",
    asAligned: "center",

    withColor: null,
    withIcon: null,
    withAnimation: null,
    withTranslation: null,

    isHidden: false,
    isDisabled: false,
    isFluid: false,
};

function getColors(colors, emphasis, hovered) {
    let colorStyle = hovered
        ? {
            background: colors.hoverBackgroundColor,
            color: colors.hoverTextColor,
        }
        : {
            background: emphasis !== "contained" ? "transparent" : colors.backgroundColor,
            color: emphasis !== "contained" ? colors.backgroundColor : colors.textColor,
        }
    if (!hovered && emphasis === "outlined")
        colorStyle.borderColor = colors.backgroundColor
    return colorStyle;
}
/**
## Notes
- The design system used for this component is fontawesome Icons
- The animation system used for this component is Framer Motion (framer-motion)
- Pass inline styles to the component to override any of the component css
- Or add custom css in overrule.scss to override the component css
- props are not being passed to the ToolbarDark. Please speak to the admin to handle any new prop.
**/
export default function ToolbarDark(props) {
    const [hovered, setHovered] = useState(false);
    //-------------------------------------------------------------------
    // 1. Set the classes
    //-------------------------------------------------------------------
    let quommonClasses = getQuommons(props);
    if (props.isCircular)
        quommonClasses.childClasses += ` is-circular`;

    quommonClasses.childClasses += ` emp-${props.asEmphasis}`;

    //-------------------------------------------------------------------
    // 2. Set the component colors
    //-------------------------------------------------------------------
    let colors = props.withColor ? getColors(props.withColor, props.asEmphasis, hovered) : {};

    //-------------------------------------------------------------------
    // 3. Translate the text objects in case their is a dictionary provided
    //-------------------------------------------------------------------
    if (
        props.withTranslation?.lang &&
        props.withTranslation.lang !== "" &&
        props.withTranslation.lang !== "en"
    ) {
        let tObj = getTranslation(props.withTranslation, "ToolbarDark");

    }

    //-------------------------------------------------------------------
    // 4. Get animation of the component
    //-------------------------------------------------------------------
    const animate = getAnimation(props.withAnimation);

    let { content } = props;

    // ========================= Render Function =================================

    return (
        <motion.div
            initial={animate.from}
            animate={animate.to}
            className={`qui ${quommonClasses.parentClasses}`}
            onMouseEnter={() => setHovered(true)}
            onMouseLeave={() => setHovered(false)}>
            <div className="backbar">
                <div className={`qui-icon`}>
                    {_.map(content, (icon, index) => {
                        return (
                            <div
                                key={index}
                                variant={props.asEmphasis}
                                color={props.asVariant}
                                style={Object.assign({}, colors, props.style)}
                            >
                                <IconLink
                                    {...props}
                                    withIcon={{ icon: icon.icon }}
                                    withLabel={{ content: icon.label, format: "caption" }}
                                />
                            </div>
                        )
                    }
                    )}
                </div>
            </div>
        </motion.div>
    );
};




