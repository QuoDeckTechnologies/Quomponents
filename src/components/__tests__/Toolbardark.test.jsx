import { shallow } from 'enzyme';
//--------------------------------------
// Import Common Tests
// -------------------------------------
import { hasValid } from "./common";
//--------------------------------------
// Import Components
// -------------------------------------
import ToolbarDark from '../Buttons/ToolbarDark/ToolbarDark.react'
describe("ToolbarDark", () => {
    // -------------------------------------
    // Run common tests
    // -------------------------------------

    const args = {
        target: ToolbarDark,
        required: {
            content: [],
            onClick: () => { },
        },
    };

    hasValid("defaults", args);

    hasValid("variants", args);
    hasValid("sizes", args);
    hasValid("padding", args);
    hasValid("alignment", args);

    hasValid("colors", args);
    hasValid("animations", args);

    hasValid("toggles", args);

    // -------------------------------------
    // Run component specific tests
    // -------------------------------------
    let component;
    const dictionary = JSON.stringify({
        hi: {
            toolbarDark: {
                content: [
                    { label: "प्रमाणपत्र" },
                    { label: "बटुआ" },
                    { label: "पुरस्कार" },
                    { label: "रिपोर्ट" },
                ]
            },
        },
    });
    beforeEach(() => {
        jest.resetAllMocks();
        component = shallow(
            <ToolbarDark
                content={[
                    {
                        icon: "fa fa-share",
                        label: "Certificate",
                        format: "caption",
                        link: "https://quodeck.com/",
                    },
                ]}

                asEmphasis="text"
                asVariant="primary"
                asSize="normal"
                asPadded="normal"
                asAligned="center"

                withColor={null}
                withAnimation={null}
                withTranslation={null}

                isDisabled={false}
                isHidden={false}
                isFluid={false}
                isCircular={true}
                onClick={() => { }} />
        );
    });
    it("should render correctly with translation",
        () => {
            component.setProps({
                withTranslation: {
                    lang: "hi",
                    tgt: "toolbarDark",
                    dictionary: dictionary,
                },
            });
            expect(component.exists()).toBe(true);
        });
});

