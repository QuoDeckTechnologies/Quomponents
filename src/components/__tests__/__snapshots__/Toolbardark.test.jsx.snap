// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarDark matches snapshot 1`] = `
<div
  className="qui qui-button"
>
  <div
    className="qui-backbar"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div
      className="qui-icon"
    />
  </div>
</div>
`;
